### ISSUE TITLE
MsgPack/YAML/JSON encoders mishandle 18446744073709551616.0

### ISSUE THREAD

**Issue by mejedi** [tarantool/tarantool#2113]
```
tarantool> string.format('%s',18446744073709551616.0)
---
- '1.844674407371e+19'
...

tarantool> 18446744073709551616.0
---
- 0
...

tarantool> msgpack.decode(msgpack.encode(18446744073709551616.0))
---
- 0
- 2
...

tarantool> json.decode(json.encode(18446744073709551616.0))
---
- 0
...
```

**Comment by mejedi** [comment]
Once this is fixed, update tests for #2112 to check boundary values.

`18446744073709551616.0 == 2.0**64`, i.e. `INT64_MAX+1`

**Comment by rtsisyk** [comment]
Duplicate of #2472
